kernel/thread.c

```diff
@@ -906,6 +906,77 @@ myst_process_t* myst_process_self(void)
     return myst_thread_self()->process;
 }
 
+void myst_shutdown_process_thread(myst_process_t* process, bool is_main_process)
+{
+    /* wait for all child threads to shutdown */
+    {
+        while (process->main_process_thread->group_next)
+        {
+            myst_sleep_msec(10, false);
+        }
+    }
+
+    myst_signal_free(process);
+
+    /* Send SIGHUP to children if we are the head of a process group ID */
+    myst_send_sighup_child_processes(process, is_main_process);
```

Comment:
Probably better to rename the method, now that this is more about reparenting than about sending SIGHUP.